### DISCUSSION TITLE
Is it possible to compile sycl host code to ARM arch?

### DISCUSSION THREAD

**Discussion by harishch4** [intel/llvm#8263] [Q&A]
So as mentioned in the title, I'm trying compile SYCL host code to ARM, my intial thought is to pass --target flag to -fsycl-host-compiler-options. However, I'm facing below error while compiling a sycl code (cuda code converted to SYCL via SYCLomatic).

`$ clang++ -fsycl -fsycl-host-compiler="clang++" -fsycl-host-compiler-options="--target=aarch64-none-eabi" vector_add.dp.cpp`

> In file included from :1:
> /tmp/vector_add.dp-header-c8ee3f.h:148:20: error: no template named 'KernelInfoData'; did you mean 'KernelInfo'?
> template <> struct KernelInfoData<'', 'Z', 'T', 'S', 'Z', '4', 'm', 'a', 'i', 'n', 'E', 'U', 'l', 'N', '4', 's', 'y', 'c', 'l', '3', '', 'V', '1', '7', 'n', 'd', '', 'i', 't', 'e', 'm', 'I', 'L', 'i', '3', 'E', 'E', 'E', 'E', ''> {
> ^~~~~~~~~~~~~~
> KernelInfo

However, if I run the same command without -fsycl-host-compiler and -fsycl-host-compiler-options, it works but for x86_64.

**Comment by bader** [comment]
@harishch4, if you can use clang++ as you host compiler, it's easier use regular mode (i.e. don't use `-fsycl-host-compiler*` options).

If you wish to use an alternative host compiler, you should know that certain features won't work, and this configuration is not tested. You haven't provided the code you tried to compile, but I suspect it uses "unnamed" kernels, which requires host compiler to support non-standard intrinsic. You can try using named kernels to fix this problem.
